File: src/main/java/space/mori/regenblock/RegenBlockLogHandler.java
package space.mori.regenblock;

import java.util.logging.*;
import org.bukkit.plugin.*;
import org.bukkit.*;
import org.bukkit.entity.*;
import java.util.*;

public class RegenBlockLogHandler
{
    private RegenBlock plugin;
    private Logger logger;
    
    public RegenBlockLogHandler(final RegenBlock plugin) {
        this.plugin = plugin;
        this.logger = Logger.getLogger("Minecraft");
    }
    
    private String buildString(final String message) {
        final PluginDescriptionFile pdFile = this.plugin.getDescription();
        return "[" + pdFile.getName() + "] (" + pdFile.getVersion() + ") " + message;
    }
    
    private String buildStringPlayer(final String message, final ChatColor color) {
        final PluginDescriptionFile pdFile = this.plugin.getDescription();
        return color + "[" + pdFile.getName() + "] " + message;
    }
    
    public void info(final String message) {
        this.logger.info(this.buildString(message));
    }
    
    public void warn(final String message) {
        this.logger.warning(this.buildString(message));
    }
    
    public void sendPlayerNormal(final Player player, final String message) {
        player.sendMessage(this.buildStringPlayer(message, ChatColor.AQUA));
    }
    
    public void sendPlayerWarn(final Player player, final String message) {
        player.sendMessage(this.buildStringPlayer(message, ChatColor.YELLOW));
    }
    
    public void sendPlayerRegionInfo(final Player player, final String regionName) {
        final String left = this.plugin.getConfiguration().getRegionLeft(regionName);
        final String right = this.plugin.getConfiguration().getRegionRight(regionName);
        final String worldName = this.plugin.getConfiguration().getRegionWorldName(regionName);
        final int respawnTime = this.plugin.getConfiguration().getRegionRespawnTime(regionName);
        final int sync = this.plugin.getConfiguration().getRegionSync(regionName);
        final int type = this.plugin.getConfiguration().getRegionType(regionName);
        final int feedbackId = this.plugin.getConfiguration().getRegionFeedbackID(regionName);
        final int alarmTime = this.plugin.getConfiguration().getRegionAlarmTime(regionName);
        final int alarmRadius = this.plugin.getConfiguration().getRegionAlarmRadius(regionName);
        final String alarmMessage = this.plugin.getConfiguration().getRegionAlarmMessage(regionName);
        this.sendPlayerNormal(player, String.valueOf(regionName) + ": [W] " + worldName + "; [L] " + left + "; [R] " + right + "; [T] " + respawnTime + "s.");
        this.sendPlayerNormal(player, "[Sync] " + sync + "; [Type] " + type + "; [Feedback ID] " + feedbackId);
        this.sendPlayerNormal(player, "Alarm [Time] " + alarmTime + "; [Radius] " + alarmRadius + "; [Message] " + alarmMessage);
    }
    
    public void listRegion(final Player player, final Set<String> listRegion) {
        for (final String regionName : listRegion) {
            final String left = this.plugin.getConfiguration().getRegionLeft(regionName);
            final String right = this.plugin.getConfiguration().getRegionRight(regionName);
            final String worldName = this.plugin.getConfiguration().getRegionWorldName(regionName);
            final int respawnTime = this.plugin.getConfiguration().getRegionRespawnTime(regionName);
            this.sendPlayerNormal(player, String.valueOf(regionName) + ": [W] " + worldName + " [L] " + left + "; [R] " + right + "; [T] " + respawnTime + "s.");
        }
    }
}

File: src/main/java/space/mori/regenblock/RegenBlockLogHandler.kt
package space.mori.regenblock

import java.util.logging.*
import org.bukkit.plugin.*
import org.bukkit.*
import org.bukkit.entity.*
import java.util.*

class RegenBlockLogHandler(private val plugin: RegenBlock) {
    private val logger: Logger = Logger.getLogger("Minecraft")

    private fun buildString(message: String): String {
        val pdFile = this.plugin.description
        return "[${pdFile.name}] (${pdFile.version}) $message"
    }

    private fun buildStringPlayer(message: String, color: ChatColor): String {
        val pdFile = this.plugin.description
        return "$color[${pdFile.name}] $message"
    }

    fun info(message: String) {
        this.logger.info(this.buildString(message))
    }

    fun warn(message: String) {
        this.logger.warning(this.buildString(message))
    }

    fun sendPlayerNormal(player: Player, message: String) {
        player.sendMessage(this.buildStringPlayer(message, ChatColor.AQUA))
    }

    fun sendPlayerWarn(player: Player, message: String) {
        player.sendMessage(this.buildStringPlayer(message, ChatColor.YELLOW))
    }

    fun sendPlayerRegionInfo(player: Player, regionName: String) {
        val left = this.plugin.configuration!!.getRegionLeft(regionName)
        val right = this.plugin.configuration!!.getRegionRight(regionName)
        val worldName = this.plugin.configuration!!.getRegionWorldName(regionName)
        val respawnTime = this.plugin.configuration!!.getRegionRespawnTime(regionName)
        val sync = this.plugin.configuration!!.getRegionSync(regionName)
        val type = this.plugin.configuration!!.getRegionType(regionName)
        val feedbackId = this.plugin.configuration!!.getRegionFeedbackID(regionName)
        val alarmTime = this.plugin.configuration!!.getRegionAlarmTime(regionName)
        val alarmRadius = this.plugin.configuration!!.getRegionAlarmRadius(regionName)
        val alarmMessage = this.plugin.configuration!!.getRegionAlarmMessage(regionName)
        this.sendPlayerNormal(
            player,
            "$regionName: [W] $worldName; [L] $left; [R] $right; [T] ${respawnTime}s."
        )
        this.sendPlayerNormal(player, "[Sync] $sync; [Type] $type; [Feedback ID] $feedbackId")
        this.sendPlayerNormal(player, "Alarm [Time] $alarmTime; [Radius] $alarmRadius; [Message] $alarmMessage")
    }

    fun listRegion(player: Player, listRegion: Set<String>) {
        for (regionName in listRegion) {
            val left = this.plugin.configuration!!.getRegionLeft(regionName)
            val right = this.plugin.configuration!!.getRegionRight(regionName)
            val worldName = this.plugin.configuration!!.getRegionWorldName(regionName)
            val respawnTime = this.plugin.configuration!!.getRegionRespawnTime(regionName)
            this.sendPlayerNormal(
                player,
                "$regionName : [W] $worldName [L] $left; [R] $right; [T]${respawnTime}s."
            )
        }
    }
}
